client/redux/auth/authService.ts

```diff
@@ -0,0 +1,53 @@
+import { deleteCookie, setCookie } from 'cookies-next';
+
+import { axios, setBearerToken } from 'shared/lib/axios';
+import { SignInUpFormValues } from 'shared/types';
+
+const signUp = async (user: SignInUpFormValues): Promise<any> => {
+  const response = await axios.post('/sign-up', user);
+  const { token, user: data } = response.data;
+
+  if (response.status === 200) {
+    setCookie('token', token);
+    setBearerToken(token);
+  }
+
+  return data;
+};
+
+const signIn = async (user: SignInUpFormValues): Promise<any> => {
+  const response = await axios.post('/sign-in', user);
+  const { token, user: data } = response.data;
+
+  if (response.status === 200) {
+    setCookie('token', token);
+    setBearerToken(token);
+  }
+
+  return data;
+};
+
+const signOut = async (): Promise<any> => {
+  const response = await axios.post('/sign-out');
+  if (response.status === 204) {
+    deleteCookie('token');
+  }
+  return response.data;
+};
+
+const hydrateUserState = async (): Promise<any> => {
+  const response = await axios.get('/auth');
+  if (response.status === 200) {
+    return response.data;
+  }
+  return 'Something went wrong';
+};
```

Comment:
Errors will be thrown in the catch section in the thunk methods

```suggestion
const hydrateUserState = async (): Promise<any> => {
  const response = await axios.get('/auth');
  return response.data
};
```